server/CGameHandler.cpp

```diff
@@ -3147,33 +3147,40 @@ bool CGameHandler::buildStructure(ObjectInstanceID tid, BuildingID requestedID,
 		processBeforeBuiltStructure(builtID);
 
 	//Take cost
-	if (!force)
-	{
+	if(!force)
 		giveResources(t->tempOwner, -requestedBuilding->resources);
-	}
 
-	//We know what has been built, appluy changes. Do this as final step to properly update town window
+	//We know what has been built, apply changes. Do this as final step to properly update town window
 	sendAndApply(&ns);
 
 	//Other post-built events. To some logic like giving spells to work gamestate changes for new building must be already in place!
-	for (auto builtID : ns.bid)
+	for(auto builtID : ns.bid)
+	{
 		processAfterBuiltStructure(builtID);
+		auto building = t->town->buildings.at(builtID);
 
+		if(building->overrideBids.empty()) //NewStructures::applyGs handles it on the client side
+			continue;
+		
+		for(auto overrideBid : building->overrideBids)
+			const_cast<CGTownInstance *>(t)->overriddenBuildings.insert(overrideBid);
```

Comment:
Should this be done during netpack apply (and not only on server side)?

Reply:
> 
> 
> Should this be done during netpack apply (and not only on server side)?

Ok, `NewStructures::applyGs` works on on the client and server side.
Also, visited bonuses should be overridden too, so I've add corresponding code.